inst/private/doctest_collect.m

```diff
@@ -24,6 +24,11 @@
   [~, ~, ext] = fileparts(what);
   if any(strcmpi(ext, {'.texinfo' '.texi' '.txi' '.tex'}))
     type = 'texinfo';
+  elseif (strcmp (ext, '.oct') && exist (what) == 3)  % .oct explicitly
+    type = 'octfile';
+  elseif (exist (what) == 3)  % .oct/.mex
+    [~, what, ~] = fileparts (what);  % strip extension if present
+    type = 'function';                % then access like any function
```

Comment:
This block (`% .oct/.mex`) matches:
1.  `doctest foo` (rather than `doctest foo.oct`).  I don't think its necessary for this.
2.  `doctest bar.mex` and `doctest bar`: I haven't tested any .mex stuff yet.

Reply:
Yes, I like this distinction between `doctest func` and `doctest func.oct`. I didn't see a clean way to do it when I was making my changes.